compiler/rustc_codegen_ssa/src/traits/asm.rs

```diff
@@ -25,9 +26,12 @@ pub enum InlineAsmOperandRef<'tcx, B: BackendTypes + ?Sized> {
         in_value: OperandRef<'tcx, B::Value>,
         out_place: Option<PlaceRef<'tcx, B::Value>>,
     },
-    Const {
+    Interpolate {
```

Comment:
Please add more doc comment explaining the significance of the difference between `Interpolate` and `Const` for backends. It begs the question for why we ever turn const operands into strings, for example.

Reply:
Added some explaination. Technically we should be able to use `Const` for integers, although I am a bit conservative here and don't want to change how existing things work.

This might also be useful to have in the future if we decided to add `interpolate "some CTFE string to be interpolated"` new type of operand.